packages/renderer/src/lib/dashboard/NewContentOnDashboardBadge.svelte

```diff
@@ -0,0 +1,71 @@
+<script lang="ts">
+import { onDestroy, onMount } from 'svelte';
+import { providerInfos } from '/@/stores/providers';
+import { notificationQueue } from '/@/stores/notifications';
+import { router } from 'tinro';
+import type { Unsubscriber } from 'svelte/motion';
+
+let providersId: string[] = [];
+let notificationCount: number = 0;
+let hasNewProviders = false;
+let hasNewNotification = false;
+let isInDashBoard = true;
+let hasNew: boolean;
+$: hasNew = !isInDashBoard && (hasNewProviders || hasNewNotification);
+
+let providersUnsubscribe: Unsubscriber;
+let notificationsUnsubscribe: Unsubscriber;
+let routerUnsubscribe: Unsubscriber;
+
+onMount(() => {
+  // if there is a new provider we display the dot
+  providersUnsubscribe = providerInfos.subscribe(updatedProviders => {
```

Comment:
I don't want to turn this into a big componentization effort... but I'm pretty sure that we're going to want similar purple notification badges in other places too, e.g. if you pull an image the Images tab would have a badge until you go to it. We don't need to implement that here or make this a full component, but maybe we can extract the purple dot and cleanup based on router path from the dashboard-specific logic of when to add it and specific path?

Reply:
I get your point, I also thought about it while implementing bc i wanted to create something customizable that could be reused but I didn't find any easy way to do it. The problem is that each route could have different and multiple ways to detect new content. And instead of adding multiple `ifs` - if dashboard subscribe to, if images subscribe to, ... - I thought that creating a custom component for each place could be easier to maintain in the future. So if we add new things to the dashboard and we want to change when the purple dot is visible, we only work on this.

The thing that multiple dot implementation can have in common is the router.subscribe and reset of the dot state, so i can create a baseComponent with it. WDYT?

Reply:
> The thing that multiple dot implementation can have in common is the router.subscribe and reset of the dot state, so i can create a baseComponent with it. WDYT?

Yeah, that's all I was thinking - a NotificationBadge component with 'I will disappear when user gets here' path property and route logic, outer dashboard component sets '/' and has logic of when to trigger. It's not much but stops us from replicating router code and UI in other instances.